extensions/wikia/FeedsAndPosts/RecentChanges.class.php

```diff
@@ -78,6 +78,10 @@ private function filterOutSmallChangesAndCleanUp( $articles ) {
 				return $article['pageid'];
 			}, $resultArticles ), DB_SLAVE );
 
+		$resultTitles = array_filter($resultTitles, function ( $title ) {
+			return $title != null;
```

Comment:
What about doing something like
```
$pageIds = [];
$result = [];

for ( $resultArticles as $article ) {
    $pageIds[] = $article['pageid'];
}

$titles = Title::newFromIds( $pageIds );

foreach ( $titles as $title ) {
    $info[] = [
         'title' => $title->getText(),
	'url' => $title->getLocalURL(),
    ];
}
```
`Title::newFromIds`  will never return null titles.

Reply:
Will `$titles` be sorted in right order then?

Reply:
True, we're operating on sorted data. In that case we could just convert the `array_map` call to a `foreach` loop & do an `isset` check there - then we won't iterate twice.